### PR TITLE
Better documentation for functional tests in Java

### LINKED ISSUE #4168: Fix JavaFunctionalTest documentation
https://www.playframework.com/documentation/2.4.x/JavaFunctionalTest

has the following problems:

* `GuiceApplicationBuilder` should be used in favour of `FakeApplication`, so this should be briefly demonstrated, with links to JavaTestingWithGuice for full documentation.
* `WithApplication` only gets a brief mention, `WithServer` and `WithBrowser` don't get mentioned at all, these should have the spotlight on this page.
* References to `fakeRequest` should be replaced with documentation on how to use `play.mvc.Http.RequestBuilder`

### PR DESCRIPTION
**Pull request by megazord**
This should fix #4168.

### CODE REVIEW

File: documentation/manual/working/javaGuide/main/tests/JavaFunctionalTest.md [hunk 1]
```diff
@@ -26,25 +26,42 @@ To run tests with an `Application`, you can do the following:
...
 
+This will ensure that a [`FakeApplication`](api/java/play/test/FakeApplication.html) will be started and stopped for each test method.
+
+## Testing with a Guice application
+
+To run tests with an `Application` [[created by Guice|JavaTestingWithGuice]], you can do the following:
+
+@[test-guiceapp](code/javaguide/tests/guice/JavaGuiceApplicationBuilderTest.java)
+
+Note that are different ways to customized the `Application` creation when using Guice to test.
```
Comment (benmccann) [inline documentation/manual/working/javaGuide/main/tests/JavaFunctionalTest.md]: "customized" should be "customize"

"Note that are" should be "Note that there are"

File: documentation/manual/working/javaGuide/main/tests/JavaFunctionalTest.md [hunk 2]
```diff
@@ -26,25 +26,42 @@ To run tests with an `Application`, you can do the following:
...
+
 ## Testing with a server
 
 Sometimes you want to test the real HTTP stack from within your test. You can do this by starting a test server:
 
 @[test-server](code/javaguide/tests/FunctionalTest.java)
 
+Just as exists a `WithApplication` class, there is also a [`WithServer`](api/java/play/test/WithServer.html) which you can extend to automatically start and stop an [`TestServer`](api/java/play/test/TestServer.html) for your tests:
```
Comment (benmccann) [inline documentation/manual/working/javaGuide/main/tests/JavaFunctionalTest.md]: "Just as exists" -> "Just as there exists"

"an TestServer" -> "a TestServer"

File: documentation/manual/working/javaGuide/main/tests/JavaFunctionalTest.md [hunk 3]
```diff
@@ -26,25 +26,42 @@ To run tests with an `Application`, you can do the following:
...
+@[test-withserver](code/javaguide/tests/ServerFunctionalTest.java)
+
 ## Testing with a browser
 
 If you want to test your application from with a Web browser, you can use [Selenium WebDriver](http://code.google.com/p/selenium/?redir=1). Play will start the WebDriver for you, and wrap it in the convenient API provided by [FluentLenium](https://github.com/FluentLenium/FluentLenium).
 
-@[with-browser](code/javaguide/tests/FunctionalTest.java)
+@[test-browser](code/javaguide/tests/FunctionalTest.java)
+
+And, of course there is the [`WithBrowser`](api/java/play/test/WithBrowser.html) class to automatically open and close a browser for each test:
```
Comment (benmccann) [inline documentation/manual/working/javaGuide/main/tests/JavaFunctionalTest.md]: "And, of course there" -> "And, of course, there"

**Comment by jroper** [comment]
@megazord Different package names, the code snippet is in `code/tests/guice`, you're pointing to `code/javaguide/tests/guice`.

**Comment by megazord** [comment]
@benmccann thanks for the corrections! I will submit a new version in a few minutes.

@jroper thanks! Looks like IntelliJ was copying the wrong reference/path.

**Comment by megazord** [comment]
Just made the corrections. 

@jroper and @benmccann, could you please take a look?

Thanks!

File: documentation/manual/working/javaGuide/main/tests/code/javaguide/tests/FunctionalTest.java [hunk 4]
```diff
@@ -20,7 +20,11 @@
     //#bad-route
     @Test
     public void testBadRoute() {
-        Result result = route(fakeRequest(GET, "/xx/Kiki"));
+        play.mvc.Http.RequestBuilder request = new play.mvc.Http.RequestBuilder()
```
Comment (benmccann) [inline documentation/manual/working/javaGuide/main/tests/code/javaguide/tests/FunctionalTest.java]: Can you import `play.mvc.Http.RequestBuilder` so you don't have to refer to it here by the fully qualified package name?

Comment (megazord) [inline documentation/manual/working/javaGuide/main/tests/code/javaguide/tests/FunctionalTest.java]: Sure! I will do it tonight.

**Comment by megazord** [comment]
@benmccann done!

**Comment by megazord** [comment]
Not sure why [the build broke](https://playframework2.ci.cloudbees.com/job/play-master-PRs/4233/console). It just shows that there is a problem with scaladoc generation:

```
[error] (SBT-Fork-Run-Protocol/compile:compileIncremental) Compilation failed
[error] (Fork-Run-Protocol/compile:doc) Scaladoc generation failed
[error] Total time: 183 s, completed Apr 13, 2015 6:47:58 PM
```
@benmccann, please, trigger a new build to see if it was some glimpse with the build.

**Comment by benmccann** [comment]
@megazord i think you can probably trigger as well. you do it by mentioning @play-pull-request-validator. and now that i've done that a new build should be starting :-)

**Comment by megazord** [comment]
Didn't know that. Nice, thanks @benmccann. :-)

**Comment by megazord** [comment]
And we have a green build. I think that now this PR is ok to be merged.

**Pull request merged**